### PR TITLE
Hide internal actors from rocket state

### LINKED ISSUE #774: Too many details about the actor system leak into rocket
Currently, we are managing `xtra::Address`es within rocket and send messages to dedicated actors upon calls to certain endpoints.

As we move away from a monolithic `{taker,maker}_cfd::Actor`, which actor to contact for a certain behaviour may change. For example, the short-lived collaborative settlement actor _could_ be spawned directly from the `maker_inc_connections::Actor` upon receiving the incoming message. That would have the advantage that we would only maintain the state about ongoing collaborative settlements once (as an `AddressMap` inside the `maker_inc_connections::Actor`).

However, changing which actor to send the `Accept` or `Reject` message to (from `maker_cfd::Actor` to `maker_inc_connections::Actor`) requires an unnecessary change to the rocket-managed state.

This highlights that too many details about the actor system are currently leaked to its users.

There are likely multiple ways of fixing this. We should consider doing something like:

```rs
impl MakerActorSystem {
	pub async fn accept_settlement(&self, order_id: OrderId) -> Result<()> {
		todo!()
	}
}
```

And `rocket.manage` the entire `MakerActorSystem`. This would allow us to make changes to the internals of the actor system without them leaking into upper layer.

### PR DESCRIPTION
**Pull request by rishflab**
Partially addresses #774 

will squash after review

### CODE REVIEW

File: daemon/src/routes_maker.rs [hunk 1]
```diff
@@ -152,27 +146,25 @@ pub struct PromptAuthentication {
 pub async fn post_cfd_action(
     id: OrderId,
     action: CfdAction,
-    cfd_actor: &State<Maker>,
+    maker_actor_system: &State<Maker>,
```
Comment (thomaseizinger) [inline daemon/src/routes_maker.rs]: Nit: I think naming this `maker` would be expressive enough.

```suggestion
    maker: &State<Maker>,
```

File: daemon/src/maker.rs [hunk 2]
```diff
@@ -294,15 +290,19 @@ async fn main() -> Result<()> {
         Poll::Ready(Some(message))
     });
 
-    tasks.add(incoming_connection_addr.attach_stream(listener_stream));
+    tasks.add(
+        maker_actor_system
+            .inc_conn_addr
+            .clone()
+            .attach_stream(listener_stream),
```
Comment (thomaseizinger) [inline daemon/src/maker.rs]:
```suggestion
        maker_actor_system.listen_on(listener)
```

I think we could hide a few more details here, that should allow us to make all fields of `MakerActorSystem` private.

**Review by thomaseizinger — APPROVED** [review]
Thank you! Some ideas but good to see this idea play out as envisioned!

**Review by thomaseizinger — APPROVED** [review]
Thanks! Nice work!

Feel free to squash everything into one commit.

**Comment by rishflab** [comment]
I think I have broken it:

```

    pub fn listen_on(&mut self, listener: TcpListener) {
        let listener_stream = futures::stream::poll_fn(move |ctx| {
            let message = match futures::ready!(listener.poll_accept(ctx)) {
                Ok((stream, address)) => {
                    maker_inc_connections::ListenerMessage::NewConnection { stream, address }
                }
                Err(e) => maker_inc_connections::ListenerMessage::Error { source: e },
            };

            Poll::Ready(Some(message))
        });

        self.tasks
            .add(self.inc_conn_addr.clone().attach_stream(listener_stream));
            ```

**Pull request merged**